### 1. Đọc bộ dữ liệu

In [4]:
import pandas as pd
df = pd.read_csv('GDPlist.csv', encoding = 'ISO-8859-1')
df

Country      Continent  GDP (millions of US$)
0     Afghanistan           Asia                  18181
1         Albania         Europe                  12847
2         Algeria         Africa                 190709
3          Angola         Africa                 100948
4       Argentina  South America                 447644
..            ...            ...                    ...
120    Uzbekistan           Asia                  45353
121     Venezuela  South America                 315841
122       Vietnam           Asia                 122722
123         Yemen         Africa                  33675
124        Zambia         Africa                  19206

[125 rows x 3 columns]

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 125 entries, 0 to 124
Data columns (total 3 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   Country                125 non-null    object
 1   Continent              125 non-null    object
 2   GDP (millions of US$)  125 non-null    int64 
dtypes: int64(1), object(2)
memory usage: 3.1+ KB


+ Dữ liệu có 3 cột, 125 dòng
+ Country, Continet là biến định tính, thang đo định danh
+ GDP là biến định lượng, thang đo tỉ lệ

### 2. Tính giá trị lớn nhất, nhỏ nhất của GDP

In [11]:
print('Giá trị GDP nhỏ nhất là', df['GDP (millions of US$)'].min(), 'millions of US$')
print('Giá trị GDP lớn nhất là', df['GDP (millions of US$)'].max(), 'millions of US$')

Giá trị GDP nhỏ nhất là 10600 millions of US$
Giá trị GDP lớn nhất là 15094025 millions of US$


### 3. Hãy cho biết xu hướng phân bố dữ liệu của GDP

In [20]:
# Trung bình
df['GDP (millions of US$)'].mean()

555427.784

In [22]:
# Trung vị
df['GDP (millions of US$)'].median()

96089.0

mean > median => Dữ liệu phân bố lệch phải

### 4. Hãy cho biết châu lục nào xuất hiện nhiều nhất

In [18]:
df['GDP (millions of US$)'].mode()
## Europe xuất hiện nhiều nhất

0    Europe
Name: Continent, dtype: object

### 5. Với mỗi châu lục, hãy tính tổng GDP; trung bình cộng GDP. Hợp nhất 2 bảng này thành một bảng duy nhất gồm 3 thông tin: Tên châu lục; Tổng GDP; TBC GDP

In [51]:
df.pivot_table(values='GDP (millions of US$)', index='Continent',aggfunc ={'sum','mean'})

mean       sum
Continent                            
Africa         6.278659e+04   1820811
Asia           6.784970e+05  24425891
Europe         5.172805e+05  19139380
North America  1.658041e+06  18238447
Oceania        5.542423e+05   1662727
South America  4.601352e+05   4141217